src/backend/services/workflow_service.py

```
@@ -481,6 +485,39 @@ async def stream_execute_only(robot_code: str) -> Generator[str, None, None]:
+                        chroma_store = KeywordVectorStore(persist_directory=settings.OPTIMIZATION_CHROMA_DB_PATH)
+                        pattern_matcher = QueryPatternMatcher(db_path=settings.OPTIMIZATION_PATTERN_DB_PATH)
+                        smart_provider = SmartKeywordProvider(
+                            library_context=library_context,
+                            pattern_matcher=pattern_matcher,
+                            vector_store=chroma_store
+                        )
+                        
+                        # Learn from the successful execution
+                        smart_provider.learn_from_execution(user_query, robot_code)
+                        logging.info("✅ Pattern learning completed - learned from PASSED test")
+                except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)